### DISCUSSION TITLE
Questions about working with round_corners() & arcs

### DISCUSSION THREAD

**Discussion by dan-p3rry** [BelfrySCAD/BOSL2#1528] [Q&A] [answered]
Two questions about the following code:

1.  I want to round the corners of a 120 degree arc.  Unfortunately round_corners() doesn't differentiate between the $fa segments of the arc and a sharp corner, so I cannot round the sharp corners with a value of r larger than $fa.  Question: is there an alternate way to round the corners?  If I was working with a 90 degree arc, I could let round_corners create the arc from a rect(), but I don't what the starting shape might be for a 120 degree wedge.

2. In the code below, I use offset_sweep to turn the arc into a nicely rounded wedge of cheese.  Then I need to open a window in the rounded face of that wedge of cheese.  If I wanted square corners I could use rotate_sweep().  I spent a lot of time in the documentation, but couldn't figure out how to open a round-corner window except the hull() of four cylinders.  I'm kind of lucky even this worked -- depending on what additional shapes are present, this may not have been an acceptable solution.  Question: can you think of another way to open the window?  Here's a picture: 
![Screenshot 2024-12-17 215828](https://github.com/user-attachments/assets/18fa914b-f473-4994-b15d-4beb0397e02e)

Thanks & happy holidays!
Dan

`
include <BOSL2/std.scad>
include <BOSL2/rounding.scad>

$fs = 0.4; $fa = 3;

diam = 151.3;
shell_th = 3.6;
axispt = [shell_th/tan(60), shell_th];

container_shell_120(80);

module container_shell_120(height) {
    rounded_arc = round_corners(r=3, arc(r=0.5*diam, angle=120, wedge=true));
    interior = offset(rounded_arc, delta=shell_th);

    difference() {
        up(0.5*height) 
            offset_sweep(rounded_arc, height=height, bottom=os_teardrop(r=3), 
                top=os_circle(r=2), anchor="origin");
        up(0.5*height + 2)
            offset_sweep(interior, height=height - 2, 
                    bottom=os_circle(r=2), top=os_circle(r=-3), anchor="origin");

        #hull_arc(height);
    }
}

module hull_arc(height) {
    // angle of sweep = 120 - 2*angle corresponding to 10mm of arc
    sweep_angle = 120 - 2*20*360/(PI*diam);     // 15 is 10mm of arc + 5mm for corner rad
    hull() {
        up(20) zrot(60 - 0.5*sweep_angle) right(0.5*diam - 5) 
                xcyl(r=10, h=70, anchor=LEFT);
        up(height - 20) zrot(60 - 0.5*sweep_angle) right(0.5*diam - 5) 
                xcyl(r=10, h=70, anchor=LEFT);
        up(20) zrot(60 + 0.5*sweep_angle) right(0.5*diam - 5) 
                xcyl(r=10, h=70, anchor=LEFT);
        up(height - 20) zrot(60 + 0.5*sweep_angle) right(0.5*diam - 5) 
                xcyl(r=10, h=70, anchor=LEFT);
    }
}
`

**Comment by dan-p3rry** [comment, marked as answer]
Answering my own question ... A one line BOSL2 module to cut out that arc shape:
`bent_cutout_mask(container_r, wall_thickness+slop, subdivide_path(rect([w, h], rounding=corner_rad), 256, closed=true));`

This has a similar disadvantage as method above, the shape being difference'd extends quite far in and out of the cylinder.

**Comment by adrianVmariano** [comment]
Why is it a disadvantage that the mask shape extends out far from the cylinder?
